pkg/kudoctl/cmd/init.go

```diff
@@ -0,0 +1,155 @@
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	manInit "github.com/kudobuilder/kudo/pkg/kudoctl/cmd/init"
+	"github.com/kudobuilder/kudo/pkg/kudoctl/kube"
+
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/kubernetes"
+)
+
+const initDesc = `
+This command installs KUDO Manager onto your Kubernetes Cluster and sets up local configuration in $KUDO_HOME (default ~/.kudo/).
+
+As with the rest of the KUDO commands, 'kudo init' discovers Kubernetes clusters
+by reading $KUBECONFIG (default '~/.kube/config') and using the default context.
+
+To set up just a local environment, use '--client-only'. That will configure
+$KUDO_HOME, but not attempt to connect to a Kubernetes cluster and install the KUDO Manager.
+
+When installing KUDO manager, 'kudo init' will attempt to install the latest released
+version. You can specify an alternative image with '--kudo-image'.
+
+To dump a manifest containing the KUDO deployment YAML, combine the
+'--dry-run' and '--debug' flags.
+`
+
+type initCmd struct {
+	out        io.Writer
+	fs         afero.Fs
+	image      string
+	dryRun     bool
+	output     string
+	version    string
+	wait       bool
+	timeout    int64
+	kubeClient kubernetes.Interface
+}
+
+func newInitCmd(fs afero.Fs, out io.Writer) *cobra.Command {
+	i := &initCmd{fs: fs, out: out}
+
+	cmd := &cobra.Command{
+		Use:   "init",
+		Short: "Initialize KUDO on both client and server",
+		Long:  initDesc,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 0 {
+				return errors.New("this command does not accept arguments")
+			}
+
+			return i.run()
+		},
+	}
+
+	f := cmd.Flags()
+	f.StringVarP(&i.image, "kudo-image", "i", "", "Override KUDO manager image and/or version")
```

Comment:
I think also specifying both image and version should probably end with an error? because it's kind of undefined...

Reply:
this